File: MadHacks-2024/madhacks/backend/backend.js
const readline = require("readline");
const Profile = require('./profile');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
});

// class Profile {
//     constructor(username, password, name) {
//         this.username = username;
//         this.password = password;
//         this.name = name;
//     }

//     getPassword() {
//         return this.password;
//     }
// }

const allUsers = new Map();
let person;


function main() {
    console.log("test");

    function prompt() {
        rl.question("Enter login, signup, or stop: ", (option) => {
            if (option === "login") {
                login();
            } else if (option === "stop") {
                rl.close();
            } else if (option === "signup") {
                createAccount();
            } else {
                console.log("Invalid option, please enter login, signup, or stop.");
                prompt();
            }
        });
    }

    prompt();
}

// UPDATE when login page
function getCorrectUsername() {
    let username;
    
    do {
      // Prompt user for a username
      username = prompt('Enter a username:'); // UPDATE when login page
      
      // Check if the username already exists
      if (!allUsers.has(username)) {
        // If username exists, notify the user and prompt again
        alert('That username does not exist, please try again.');
      }
    } while (!allUsers.has(username));  // Keep asking until username is correct
  
    // Return the correct username
    return username;
  }

// UPDATE when login page
function login() {
    username = getCorrectUsername();
    if(allUsers.get(username).getPassword() === password){
        console.log("Login successful");
        person = allUsers.get(username);
    }
    else{
        console.log("error");
    }

    //rl.question("Enter username: ", (username) => {
        rl.question("Enter password: ", (password) => {
            if (allUsers.has(username) && allUsers.get(username).getPassword() === password) {
                console.log("success");
            } else {
                console.log("error");
            }
            console.log("logged in");
        

            main();  // Go back to main prompt
        });
   // });
}

function isUsernameTaken(username) {
    return allUsersMap.has(username);
  }

// UPDATE when login page
function getUniqueUsername() {
    let username;
    
    do {
      // Prompt user for a username
      username = prompt('Enter a username:'); // UPDATE when login page
      
      // Check if the username already exists
      if (isUsernameTaken(username)) {
        // If username exists, notify the user and prompt again
        alert('Username is already taken, please choose another.');
      }
    } while (isUsernameTaken(username));  // Keep asking until username is available
  
    // Return the unique username
    return username;
  }

  // UPDATE when login page
function createAccount() {
    username = getUniqueUsername();
    password = prompt('Enter a password:');
    name = prompt('Enter your name: ')
    person = new Profile(username, password, name)
    allUsers.set(username, person);
    console.log("Account created successfully");
    
   // rl.question("Enter username: ", (username) => {

     //   rl.question("Enter password: ", (password) => {
       //     rl.question("Enter your name: ", (name) => {
        //       allUsers.set(username, new Profile(username, password, name));
    //            console.log("Account created successfully");
      //          main();  // Go back to main prompt
           // });
    //    });
  //  });
}

main();

File: MadHacks-2024/madhacks/backend/backend.java


// // import static org.junit.jupiter.api.Assertions.assertEquals;

// // import org.junit.jupiter.api.Test;
// import java.util.HashMap;
// import java.util.Map;
// import java.util.LinkedList;
// import java.util.List;
// import java.util.Scanner;

// public class Main {
//      public static Map<String, Profile> allUsers = new HashMap<String, Profile>();

//      public static void main(String[] args) {

//           // allUsers = new HashMap<String, Profile>();
//           Scanner scanner = new Scanner(System.in);
//           System.out.println("test");

//           while (true) {
//                System.out.println("Enter login or signup: ");
//                String option = scanner.nextLine();
//                if (option.equals("login")) {
//                     login(scanner);
//                } else if (option.equals("stop")) {
//                     break;
//                } else {
//                     createAccount(scanner);
//                }

//           }
//           scanner.close();

//      }

//      public static void login(Scanner scanner) {
  
//           System.out.println("Enter username:");

//           String input = scanner.nextLine();
//           System.out.println("Enter password");
//           String input2 = scanner.nextLine();
//           if (allUsers.containsKey(input) && allUsers.get(input).getPassword().equals(input2)) {
//                System.out.println("success");
//           } else {
//                System.out.println("error");
//           }
//           System.out.println("logged in");
         
//      }

//      public static void createAccount(Scanner scanner) {
      
//           System.out.println("Enter username: ");
//           String user = scanner.nextLine();
//           System.out.println("Enter password: ");
//           String pass = scanner.nextLine();
//           System.out.println("Enter your name: ");
//           String name = scanner.nextLine();
//           allUsers.put(user, new Profile(user, pass, name));


//      }

// }

